lib/internal/child_process.js

```diff
@@ -763,10 +763,15 @@ function _validateStdio(stdio, sync) {
         type: 'inherit',
         fd: i
       });
-    } else if (typeof stdio === 'number' || typeof stdio.fd === 'number') {
+    } else if (typeof stdio === 'number') {
       acc.push({
         type: 'fd',
-        fd: stdio.fd || stdio
```

Comment:
May be we could just do `typeof stdio === 'number' ? stdio : stdio.fd`? I don't like duplicating the bodies.